src/operators.jl

```diff
@@ -858,3 +855,23 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, vs::Vector{U})  where U <: Integer
+    vs = sort(vs, rev=true)
+    v0 = vs[end]
+    for v in vs[1:end-1]
+        for u in inneighbors(g, v)
+            if !(u in vs)
```

Comment:
`searchsorted` might work better here since you've already ensured `vs` is (reverse) sorted. If that doesn't work, consider making a set corresponding to `vs` so we have O(1) membership checking.